deeper/plan.md

```diff
@@ -0,0 +1,182 @@
+
+The EDSL and PLAN proper are tightly intertwined:
+
+1. Within a law body, you can always drop back into regular PLAN either by using the `Const` construct or automatically when a part of the law body doesn't have meaning in the EDSL.
+2. Conversely, PLAN switches into the EDSL whenever a law has been applied to enough arguments.
+
+This dual-language structure allows PLAN to maintain a simple core while providing more expressive power for defining functions.
+
+## Reflexivity in PLAN
+
+In PLAN, any program can deconstruct and analyze any terms it encounters during execution. This means that PLAN programs have the ability to:
+
+1. Inspect the structure of data and code
+2. Make decisions based on the form of expressions
+3. Modify or generate new code dynamically
+
+This level of reflexivity is not common in many programming languages or evaluation models.
+
+## PLAN as a Comprehensive Data Structure
+
+While it's not entirely accurate to call PLAN a format for human-readable executables or binaries, this perspective highlights PLAN's unique position as a comprehensive data structure that bridges multiple domains of computation and storage.
+
+### Balancing Multiple Concerns
+
+PLAN is designed to strike a balance between several aspects of computation and data representation:
+
+1. **Human Readability**: Unlike typical low-level formats, PLAN maintains a degree of readability that allows developers to inspect and understand "binaries" directly.
+2. **Functional Compile Target**: PLAN serves as an excellent target for functional language compilers, capturing complex semantics in a simple form.
+3. **Efficient Memory Representation**: The structure of PLAN is designed to map well to in-memory data structures, allowing for efficient execution.
+4. **Effective On-Disk Representation**: PLAN's format is also suitable for direct storage, enabling persistent storage without conversion.
+5. **Closure Representation**: PLAN stores functions together with their environments, effectively representing closures and eliminating the need for free variables or assumed environments.
+
+## Clarification on Syntax and Semantics
+
+In discussions of PLAN, there may be some confusion regarding the syntax and semantics of certain constructs. It's important to clarify that:
+
+- `{}` is used to denote a law (function definition) in PLAN. It's not a general-purpose list constructor.
+- `()` is used for function application, but it's also used more generally to construct applications (which can be thought of as pairs).
+
+Neither `{}` nor `()` should be thought of as general "lists of values". Instead:
+
+- `{n a b}` represents a law with name `n`, arity `a`, and body `b`.
+- `(f x)` represents the application of `f` to `x`, which could be function application if `f` is a function, or could just be a pair if `f` is not a function.
```

Comment:
If `f` is not a function then `(f x)` will crash immediately upon construction. (I include both laws, primops and unsaturated apps when i say "function" here.)

Reply:
resolved in https://github.com/deathtothecorporation/plunder-docs/pull/21/commits/df86146a0fefcb51eed38b523d477301b1c09e75